src/mcts/node.h

```diff
@@ -164,15 +170,24 @@ class Node {
   float GetM() const { return m_; }
 
   // Returns whether the node is known to be draw/lose/win.
-  bool IsTerminal() const { return terminal_type_ != Terminal::NonTerminal; }
+  bool IsTerminal() const {
+    return terminal_type_ == Terminal::Terminal ||
+           terminal_type_ == Terminal::Tablebase;
+  }
   bool IsTbTerminal() const { return terminal_type_ == Terminal::Tablebase; }
+  bool CantWin() const { return terminal_type_ == Terminal::CantWin; }
+  bool CantLose() const { return terminal_type_ == Terminal::CantLose; }
   uint16_t GetNumEdges() const { return edges_.size(); }
 
   // Makes the node terminal and sets it's score.
   void MakeTerminal(GameResult result, float plies_left = 0.0f,
                     Terminal type = Terminal::Terminal);
   // Makes the node not terminal and updates its visits.
   void MakeNotTerminal();
+  // Indicates the node cannot win (e.g., a child can draw).
+  void MakeCantWin() { terminal_type_ = Terminal::CantWin; }
+  // Indicates the node cannot lose (i.e., the best child cannot win).
+  void MakeCantLose() { terminal_type_ = Terminal::CantLose; }
```

Comment:
@borg323 suggested `p->MarkCantWin()` as a better verb/method name. Alternatively, does it make sense to have a single method that takes a `Node::Terminal` parameter, e.g., `p->SetTerminalType(Node::Terminal::CantWin)` although the caller could pass in any valid type… which might be okay?

Reply:
I switched it to `RemoveOutcome(GameResult)` similar to your suggestion and matching existing `MakeTerminal(GameResult, …)`